docs/proposals/templates-v2.md

```diff
@@ -0,0 +1,334 @@
+4. Manually enter the template contents you want
+
+#### `Markdown Notes: New Note` (Keyboard)
+
+1. Navigate to the directory where you want the note
+2. `Shift+⌘P` to open the command pallette
+3. Type `New Note` until it appears in the list. Press `Enter/Return` to select it.
+4. Enter a name for the note
+5. Manually enter the template contents you want
+
+#### Foam template note creation (Keyboard)
+
+1. `Shift+⌘P` to open the command pallette
+2. Type `Create New Note From Template` until it appears in the list. Press `Enter/Return` to select it.
+3. Use the arrow keys (or type the template name) to select the template. Press `Enter/Return` to select it.
+4. Modify the filepath to match the desired directory + filename. Press `Enter/Return` to select it.
+
+All of these steps are far too cumbersome. And only the last one allows the use of templates.
+
+### Templating of daily notes
+
+Currently `Open Daily Note` opens an otherwise empty note, with a title defined by the `foam.openDailyNote.titleFormat` setting.
+Daily notes should be able to be fully templated as well.
+
+### Templating of filepaths
+
+As discussed in ["Template the filepath in `openDailyNote`"](https://github.com/foambubble/foam/issues/523), it would be useful to be able to specify the default filepaths of templates. For example, many people name their note filepaths in a way that includes a timestamp.
+
+## Goal / Philosophy
+
+In a sentence: **Creating a new note should be a single button press and should use templates.**
+
+## Proposal
+
+1. Add a new `Foam: Create New Note` that is the streamlined counterpart to the more flexible `Foam: Create New Note From Template`
+2. Use templates everywhere
+3. Add metadata into the actual templates themselves in order to template the filepaths themselves.
+
+### Summary
+
+This can be done through a series of changes to the way that templates are implemented:
+
+1. Add a `${title}` template variable
+2. Add a `Foam: Create New Note` command, `foam.defaultNoteTemplate` setting, and hotkey
+3. Change missing wikilinks to use the default template
+4. Add a metadata section to templates
+5. Add a `daily` note templates
+6. Eliminate all `foam.openDailyNote` settings
+
+I've broken it out into these steps to show that the overall proposal can be implemented piecemeal in independent PRs that build on one another.
+
+### Add a `${title}` template variable
+
+When you use `Markdown Notes: New Note`, and give it a title, the title is formatted as a filename and used as the title in the resulting note.
+
+**Example:**
+
+Given the name `Living in a dream world` to `Markdown Notes: New Note`, the filename is `living-in-a-dream-world.md` and the file contents are:
+
+```markdown
+# Living in a dream world
+```
+
+When creating a note from a template in Foam, you should be able to use a `${title}` variable. If the template uses the `${title}` variable, the user will be prompted for a title when they create a note from a template.
```

Comment:
In an earlier draft, I had defined a `${titleSlugified}` in `filepath`, since any title would have to be sanitized.
Then my thought was that the contents of the `filepath` will always need to be sanitized/normalized, so we don't need `${titleSlugified}` and the case of adding `${title}` in `filepath` would already be covered.

That said, having `${titleSlugified}` would allow us to separate the logic that does the slugify-ing from the logic that does the sanitizing, so it likely makes sense to have it. 

I'll add it back in.

(And people could use `${titleSlugified}` anywhere they liked, if there were other uses for it)

Reply:
I like this idea. I've added the idea of `${titleSlug}` to the Extensions section.